src/daemon/Display.cpp

```diff
@@ -309,17 +309,23 @@ namespace SDDM {
     }
 
     void Display::login(QLocalSocket *socket,
-                        const QString &user, const QString &password,
-                        const Session &session) {
+                        const QVariantMap &args) {
         m_socket = socket;
 
+        QString user = args[QStringLiteral("user")].toString();
```

Comment:
Should IMO verify that all needed keys are present

Reply:
I can't concur on that. `password` and `session*` keys may not be present (and actually aren't present in our case!). The only thing that is required is `user`, but I'm also not sure about that. Even if missing, the process would just fail at the PAM stage in this will cause the same `loginFailed` signal.